sr_utilities_common/src/sr_utilities_common/aws_manager.py

```diff
@@ -75,62 +73,187 @@ def get_bucket_structure_with_prefix(self, bucket_name, prefix):
+        if aws_subfolder:
+            if aws_subfolder[-1] != "/":
+                aws_subfolder += "/"  # Needed for formatting
+            bucket_data = self._client.list_objects(Bucket=aws_bucket, Prefix=aws_subfolder)
+        else:
+            bucket_data = self._client.list_objects(Bucket=aws_bucket)
+
+        filenames = []
+        for content in bucket_data['Contents']:
+            if aws_subfolder:
+                filenames.append(content['Key'][len(aws_subfolder):])
+            else:
+                filenames.append(content['Key'])
+
+        return filenames
+
+    def _prepare_structure_upload(self, files_base_path, files_folder_path, file_names, bucket_subfolder):
         self.file_full_paths = []
         self.aws_paths = []
         for file_name in file_names:
-            self.file_full_paths.append("{}/{}/{}".format(files_base_path, files_folder_path, file_name))
-            self.aws_paths.append("{}/{}".format(files_folder_path, file_name))
+            self.file_full_paths.append(f"{files_base_path}/{files_folder_path}/{file_name}")
+            if bucket_subfolder:
+                self.aws_paths.append(f"{bucket_subfolder}/{file_name}")
+            else:
+                self.aws_paths.append(f"{files_folder_path}/{file_name}")
 
-    def download(self, bucket_name, files_base_path, files_folder_path, file_names):
-        self._prepare_structure(bucket_name, files_base_path, files_folder_path, file_names)
-        directory = "{}/{}".format(files_base_path, files_folder_path)
+    def _prepare_structure_download(self, files_base_path, files_folder_path, file_names, bucket_subfolder):
+        self.file_full_paths = []
+        self.aws_paths = []
+        for file_name in file_names:
+            self.file_full_paths.append(f"{files_base_path}/{files_folder_path}/{file_name}")
+            if bucket_subfolder:
+                self.aws_paths.append(f"{bucket_subfolder}/{file_name}")
+            else:
+                self.aws_paths.append(file_name)
+
+    def download(self, bucket_name, files_base_path, files_folder_path, file_names, bucket_subfolder):
+        self._prepare_structure_download(files_base_path, files_folder_path, file_names, bucket_subfolder)
+        downloadSucceded = False
+        directory = os.path.join(files_base_path, files_folder_path)
         if not os.path.exists(directory):
             os.makedirs(directory)
-        downloadSucceded = False
-        for self.file_full_path, self.aws_path in zip(self.file_full_paths, self.aws_paths):
+        for file_full_path, aws_path in zip(self.file_full_paths, self.aws_paths):
             try:
-                self._client.download_file(bucket_name, self.aws_path, self.file_full_path)
+                directory = os.path.dirname(file_full_path)
+                if not os.path.exists(directory):
+                    os.makedirs(directory)
+                rospy.loginfo(f"{directory} \n{file_full_path}")
```

Comment:
why this? if you want to keep this, add more description of the purpose in the console